src/data/data.js

```diff
@@ -0,0 +1,108 @@
+import { RANKS, UPGRADE_TYPES, FACTIONS, UNIT_TYPES } from "../util/constants";
```

Comment:
Data is now a json file so I can use constants!

Reply:
I assume you mean it's now a javascript file 😅 Architectural note, if you needed to share those values across platforms of potentially different languages, you could keep it as json and have a js file in this project that takes that in and converts it to a js object to do exactly what you're doing here!

Reply:
LOL correct. and good note which may be relevant if I ever get to a backend :D